src/lambda/wizards/samDeployWizard.ts

```diff
@@ -299,35 +299,83 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
     /**
      * Retrieves an S3 Bucket to deploy to from the user.
      *
+     * @param selectedRegion Selected region for S3 client usage
      * @param initialValue Optional, Initial value to prompt with
      *
      * @returns S3 Bucket name. Undefined represents cancel.
      */
     public async promptUserForS3Bucket(selectedRegion: string, initialValue?: string): Promise<string | undefined> {
-        const inputBox = input.createInputBox({
+        const loadingBuckets: string = localize('AWS.samcli.deploy.s3bucket.picker.loading', 'Loading S3 buckets...')
+        const noBuckets: string = localize('AWS.samcli.deploy.s3bucket.picker.noBuckets', 'No buckets found.')
+        const bucketError: string = localize(
+            'AWS.samcli.deploy.s3bucket.picker.error',
+            'There was an error loading S3 buckets.'
+        )
+
+        const quickPick = picker.createQuickPick<vscode.QuickPickItem>({
             buttons: [this.helpButton, vscode.QuickInputButtons.Back],
             options: {
                 title: localize(
                     'AWS.samcli.deploy.s3Bucket.prompt',
                     'Enter the AWS S3 bucket to which your code should be deployed'
```

Comment:
Because the gesture type is changing, consider a text revision along the lines of "Select an AWS S3 Bucket to deploy code to"